### ISSUE TITLE
Jenkins Cucumber Reports plugin no longer generates report with v5.6.0 using Cucumber JVM 1.2.5 due to issue with cucumber json report file

### ISSUE THREAD

**Issue by djmonki** [jenkinsci/cucumber-reports-plugin#405]
### Version report

Jenkins and plugins versions report:
Jenkins 2.303.1
Cucumber reports Plugin: 5.6.0
Cucumber JVM 1.2.5

- What Operating System are you using (both controller, and any agents involved in the problem)?

```
Amazon Linux 2
```

### Reproduction steps

- Running DSL scripted Jenkins job using Cucumber Reports plugin, report generation fails since moving to 5.6.0. Last successful generation was with version 5.5.4 of the Jenkins plugin, where Cucumber Report is generated as part of a stage in DSL pipeline:

```
script {
  dir('ams-functional-test') {
    cucumber buildStatus: 'UNSTABLE',
       reportTitle: 'API',
       classifications: [
          [key: 'API', value: 'REST']
        ],
        jsonReportDirectory: './cucumber',
        fileIncludePattern: 'cucumber-*.json',
        trendsLimit: 30
}
```

### Results

Expected result:

Cucumber Report to be generated from cucumber-output.json

Actual result:

No report generated
Jenkins Console Output
```
[CucumberReport] Using Cucumber Reports version 5.6.0
[CucumberReport] JSON report directory is "./cucumber"
[CucumberReport] Copied 0 properties files from workspace "/home/jenkins/agent/workspace/workflow/FT/FunctionalTest/FunctionalTestDeployAndRun/functional-test/cucumber" to reports directory "/var/lib/jenkins/jobs/workflow/jobs/FT/jobs/FunctionalTest/jobs/FunctionalTestDeployAndRun/builds/78/cucumber-html-reports_20372bca-70bb-35b9-935c-76278f7463e8/.cache"
[CucumberReport] Copied 1 files from workspace "/home/jenkins/agent/workspace/workflow/FT/FunctionalTest/FunctionalTestDeployAndRun/functional-test/cucumber" to reports directory "/var/lib/jenkins/jobs/workflow/jobs/FT/jobs/FunctionalTest/jobs/FunctionalTestDeployAndRun/builds/78/cucumber-html-reports_20372bca-70bb-35b9-935c-76278f7463e8/.cache"
[CucumberReport] Processing 1 json files:
[CucumberReport] /var/lib/jenkins/jobs/workflow/jobs/FT/jobs/FunctionalTest/jobs/FunctionalTestDeployAndRun/builds/78/cucumber-html-reports_20372bca-70bb-35b9-935c-76278f7463e8/.cache/cucumber-output.json
[CucumberReport] Adding 1 classification(s)
[CucumberReport] Adding classification - API -> REST
[CucumberReport] Missing report result - report was not successfully completed
[CucumberReport] Build status is changed to UNSTABLE
```

- Actual Cucumber Report
```
net.masterthought.cucumber.ValidationException: File '/var/lib/jenkins/jobs/workflow/jobs/FT/jobs/FunctionalTest/jobs/FunctionalTestDeployAndRun/builds/78/cucumber-html-reports_20372bca-70bb-35b9-935c-76278f7463e8/.cache/cucumber-output.json' is not proper Cucumber report!
	at net.masterthought.cucumber.ReportParser.parseForFeature(ReportParser.java:104)
	at net.masterthought.cucumber.ReportParser.parseJsonFiles(ReportParser.java:72)
	at net.masterthought.cucumber.ReportBuilder.generateReports(ReportBuilder.java:97)
	at net.masterthought.jenkins.CucumberReportPublisher.generateReport(CucumberReportPublisher.java:518)
	at net.masterthought.jenkins.CucumberReportPublisher.perform(CucumberReportPublisher.java:391)
	at jenkins.tasks.SimpleBuildStep.perform(SimpleBuildStep.java:123)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:100)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:70)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.ArrayList<java.lang.Object>` from Object value (token `JsonToken.START_OBJECT`)
 at [Source: (InputStreamReader); line: 271, column: 15] (through reference chain: java.lang.Object[][0]->net.masterthought.cucumber.json.Feature["elements"]->java.lang.Object[][1]->net.masterthought.cucumber.json.Element["steps"]->java.lang.Object[][9]->net.masterthought.cucumber.json.Step["comments"]->java.util.ArrayList[0])
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1601)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1375)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1280)
	at com.fasterxml.jackson.databind.DeserializationContext.extractScalarFromObject(DeserializationContext.java:872)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseString(StdDeserializer.java:1281)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:216)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:182)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:25)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:138)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:324)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:187)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:214)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:24)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:138)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:324)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:187)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:214)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:24)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:138)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:324)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:187)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:214)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:24)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4593)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3561)
	at net.masterthought.cucumber.ReportParser.parseForFeature(ReportParser.java:93)
	... 13 more
```

**Comment by djmonki** [comment]
Have investigated and isolated issue to the following:

Gherkin Cucumber Feature file is asserting a table, where some cells are blank (null/empty)
```
And the following Resubmitted information is displayed:
    | Resubmitted | Resubmitted OCDA Ref | Application Document | Decision Document |
    | Yes         | OCDA50               | Application_Document | Decision_Document |
    |             | OCDA999              |                      |                   |
```

The json report file has this:
```
"line": 1912,
            "name": "the following Renewal information is displayed:",
            "match": {
              "location": "ViewApplicationSteps.theFollowingRenewalInformationIsDisplayed(DataTable)"            },
            "rows": [
              {
                "cells": [
                  "Renewal",
                  "Renewal OCDA Ref",
                  "Auth End Date",
                  "Application Document",
                  "Decision Document"                ],
                "line": 1913              },
              {
                "cells": [
                  "Yes",
                  "OCDA50",
                  "14-Nov-2020",
                  "Application_Document",
                  "Decision_Document"                ],
                "line": 1914              },
              {
                "cells": [
                  "",
                  "OCDA999",
                  "",
                  "",
                  ""                ],
```

The issue:
```
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.ArrayList<java.lang.Object>` from Object value (token `JsonToken.START_OBJECT`)
 at [Source: (InputStreamReader); line: 271, column: 15] (through reference chain: java.lang.Object[][0]->net.masterthought.cucumber.json.Feature["elements"]->java.lang.Object[][1]->net.masterthought.cucumber.json.Element["steps"]->java.lang.Object[][9]->net.masterthought.cucumber.json.Step["comments"]->java.util.ArrayList[0])
```

seems to be related to the cucumber-report.json:
```
"line": 1912,
            "name": "the following Renewal information is displayed:",
            "match": {
              "location": "ViewApplicationSteps.theFollowingRenewalInformationIsDisplayed(DataTable)"            },
            "rows": [
              {
                "cells": [
                  "Renewal",
                  "Renewal OCDA Ref",
                  "Auth End Date",
                  "Application Document",
                  "Decision Document"                ],
                "line": 1913              },
              {
                "cells": [
                  "Yes",
                  "OCDA50",
                  "14-Nov-2020",
                  "Application_Document",
                  "Decision_Document"                ],
                "line": 1914              },
              {
                "cells": [
                  "",
                  "OCDA999",
                  "",
                  "",
                  ""                ],
```

Looks like the de-serialisation issue is related to the plugin  can no longer cope with "",
It wants it to be populated ?

**Comment by damianszczepanik** [comment]
@jfougere another problem with comment after your commit ?

**Comment by MattLud** [comment]
To confirm, I imagine this issue is fixed by release 5.6.1 - https://github.com/jenkinsci/cucumber-reports-plugin/releases/tag/cucumber-reports-5.6.1 -  right @jfougere @damianszczepanik ?
